### DISCUSSION TITLE
How can I run Lighthouse audits in a ReactJS application, similar to using web-vitals for onCLS, onLCP, etc.

### DISCUSSION THREAD

**Discussion by basit-qayoom** [GoogleChrome/lighthouse#16132] [Q&A]
Is there a way to run Lighthouse audits in a ReactJS application similar to how we use the web-vitals npm module for onCLS, onLCP, etc.?

**Comment by brendankenny** [comment]
Not really. `web-vitals` instruments the page using JavaScript APIs to observe how the page loads. Lighthouse instruments Chrome itself to observe page loads, so can't be run within the page.

Are you looking for more than what `web-vitals` provides?

**Reply by basit-qayoom** [reply]
@brendankenny I’ve noticed a discrepancy between the CLS values reported by Web Vitals and Lighthouse. I’m currently integrating Web Vitals into my app to monitor all Core Web Vitals metrics, but I’m finding that the values from Web Vitals and Lighthouse don’t match. Any insights into this would be helpful.
